### PR TITLE
Created an automated weather forecast UI section in EventScreen

### PR DESCRIPTION
**Pull request by arnaudlvq**
#### Overview
This PR introduces the UI implementation for Gregory's work on integrating weather API calls. The app handles three scenarios:
1. The event is a past event: no weather forecast is available.
2. The event is in more than 10 days: no weather forecast is available at the moment.
3. Loading state: displays the event in a card, one per day, sorted by date, with the forecast icon, wind speed, and temperature.

The implementation uses `HorizontalPager` to achieve a carousel-like UX design, providing an intuitive way for users to navigate through weather forecasts, displaying one day at a time. The paging is inspired from this [HorizontalPager documentation](https://developer.android.com/develop/ui/compose/layouts/pager?hl=fr#add-page).

### CODE REVIEW

File: app/src/main/java/com/monkeyteam/chimpagne/ui/utilities/WeatherPager.kt [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+  val today = LocalDate.now()
+  val eventStartDate = event.startsAt().toInstant().atZone(ZoneId.systemDefault()).toLocalDate()
+  val eventEndDate = event.endsAt().toInstant().atZone(ZoneId.systemDefault()).toLocalDate()
+
+  val targetStartDate = maxOf(today, eventStartDate)
+
+  LaunchedEffect(event) {
+    if (eventEndDate.isBefore(today)) {
+      message = context.getString(R.string.noforecast_alreadyoccured)
+    } else if (targetStartDate.isAfter(today.plusDays(10))) {
```
Comment (Athanaze) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/utilities/WeatherPager.kt:67]: magic numbers like this are a bad practice

**Review by QuicheLorraine3600 — APPROVED** [review]
Nice job, the changes requested in #231 have been addressed !

File: app/src/main/java/com/monkeyteam/chimpagne/ui/utilities/WeatherPager.kt [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+      }
+    }
+  }
+}
+
+@Composable
+fun WeatherCard(weather: Weather) {
+  val configuration = LocalConfiguration.current
+  val locale = configuration.locales[0]
+  val formatter = DateTimeFormatter.ofPattern("EEEE dd MMMM yyyy", locale)
```
Comment (Athanaze) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/utilities/WeatherPager.kt:142]: We should add in the PRD something about localisation for dates like this.
Indeed, there are many instances like this in the current codebase, with the date format being not really flexible

File: app/src/main/java/com/monkeyteam/chimpagne/ui/utilities/WeatherPager.kt [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+            verticalArrangement = Arrangement.spacedBy(8.dp)) {
+              Text(
+                  text = formattedDate,
+                  style = ChimpagneTypography.titleSmall.copy(fontWeight = FontWeight.Bold))
+              Row(
+                  verticalAlignment = Alignment.CenterVertically,
+                  horizontalArrangement = Arrangement.spacedBy(16.dp)) {
+                    Box(modifier = Modifier.size(56.dp).clip(CircleShape)) {
+                      Image(
+                          painter = rememberAsyncImagePainter("https:" + weather.weatherIcon),
```
Comment (Athanaze) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/utilities/WeatherPager.kt:160]: inconsistent string concatenation style with respect to line 168 for example

**Review by Athanaze — APPROVED** [review]
Overall, while being pretty euro-centric (with the units and the date format) this style of hard-coded formatting of strings is consistent with the rest of the app, so I can't really complain too much.

**Pull request merged**